src/test/regress/flaky_tests.md

```diff
@@ -0,0 +1,227 @@
+# How to fix flaky tests
+
+There's three different causes of flaky tests.
+1. Tests that don't make sure output is consistent, i.e. a bug in our tests
+2. Bugs in our testing infrastructure
+3. Bugs in Citus itself
+
+All of these impact the happiness of our developers, because we have to rerun
+tests to make them pass. But apart for developer happiness 3 also impacts our
+users, and by ignoring flaky tests we can miss problems for our users.
+
+## Easy fixes
+
+The following types of issues all fall within the category 1: bugs in our tests.
+
+### Expected records but different order
+
+**Issue**: A query returns the right result, but they are in a different order
+than expected by the output.
+
+**Fix**: Add an extra column to the ORDER BY clause to the query to make the
+output consistent
+
+**Example**
+```diff
+  8970008 | colocated_dist_table                   | -2147483648   | 2147483647    | localhost |    57637
+  8970009 | colocated_partitioned_table            | -2147483648   | 2147483647    | localhost |    57637
+  8970010 | colocated_partitioned_table_2020_01_01 | -2147483648   | 2147483647    | localhost |    57637
+- 8970011 | reference_table                        |               |               | localhost |    57637
+  8970011 | reference_table                        |               |               | localhost |    57638
++ 8970011 | reference_table                        |               |               | localhost |    57637
+ (13 rows)
+```
+
+**Example fix**:
+
+```diff
+-ORDER BY logicalrelid, shardminvalue::BIGINT;
++ORDER BY logicalrelid, shardminvalue::BIGINT, nodeport;
+```
+
+
+### Expected logs but different order
+
+
+**Issue**: The logs in the regress output are displayed in a different order
+than what the output file shows
+
+**Fix**: It's simple: don't log these things during the test. There's two common
+ways of achieving this:
+1. If you don't care about the logs for this query at all, then you can change
+   the log `VERBOSITY`
+2. If this is are logs created by `citus.log_remote_commands` and you care about
+   some of the remote commands being as expected, but not the ones in question.
+   Then you can use `citus.grep_remote_commands` to only display the commands
+   that you care about.
+
+
+
+**Example 1**:
```

Comment:
Very NIT: Can you make the Example/Fix pair consistent across the file.
e.g. example 1/fix 1; example of issue 1/example of fix 1